rcl/src/rcl/init_options.c

```diff
@@ -34,6 +34,22 @@ rcl_get_zero_initialized_init_options(void)
   };  // NOLINT(readability/braces): false positive
 }
 
+/// Initialize given init_options with the default values and zero-initialize implementation.
+RCL_LOCAL
+rcl_ret_t
+_rcl_init_options_init(rcl_init_options_t * init_options, rcl_allocator_t allocator)
```

Comment:
```suggestion
_rcl_init_options_zero_init(rcl_init_options_t * init_options, rcl_allocator_t allocator)
```

(or something like that, just to distinguish it)

Reply:
Thanks for your quick response and the suggestion.

I have done `valgrind` for this file by `valgrind --leak-check=full ./test_init__rmw_fastrtps_cpp`, and got the same result that not leaking memory as the commit of the master branch.

<details><summary> valgrind log </summary>
<p>

```shell
[----------] Global test environment tear-down
[==========] 11 tests from 1 test suite ran. (418 ms total)
[  PASSED  ] 11 tests.
==2880876== 
==2880876== HEAP SUMMARY:
==2880876==     in use at exit: 232 bytes in 5 blocks
==2880876==   total heap usage: 2,673 allocs, 2,668 frees, 987,584 bytes allocated
==2880876== 
==2880876== LEAK SUMMARY:
==2880876==    definitely lost: 0 bytes in 0 blocks
==2880876==    indirectly lost: 0 bytes in 0 blocks
==2880876==      possibly lost: 0 bytes in 0 blocks
==2880876==    still reachable: 232 bytes in 5 blocks
==2880876==         suppressed: 0 bytes in 0 blocks
==2880876== Reachable blocks (those to which a pointer was found) are not shown.
==2880876== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==2880876== 
==2880876== For lists of detected and suppressed errors, rerun with: -s
==2880876== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

</p>
</details>

Reply:
Perfect, thank you so much for doing that additional test.  It just gives us a bit more confidence that we didn't miss something.